src/app/server/Mdns.cpp

```diff
@@ -176,6 +254,24 @@ void StartServer()
     }
 }
 
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+CHIP_ERROR GenerateRotatingDeviceId(char rotatingDeviceIdHexBuffer[], size_t & rotatingDeviceIdHexBufferSize)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    char serialNumber[chip::DeviceLayer::ConfigurationManager::kMaxSerialNumberLength + 1];
+    size_t serialNumberSize  = 0;
+    uint16_t lifetimeCounter = 0;
+    SuccessOrExit(error =
```

Comment:
nit: Using return macros instead of exit macros may make this code slightly easier to read (avoids one extra error variable)

Reply:
this code was simply moved here from another location. I plan to bring this feature in line with the others (in terms of config and other conventions) in a subsequent PR